sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2171,11 +2168,7 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
   }
 
   def noSuchElementExceptionError(key: String): Throwable = {
-    new SparkException(
-      errorClass = "_LEGACY_ERROR_TEMP_2221",
-      messageParameters = Map(
-        "key" -> key),
-      cause = null)
+    new NoSuchElementException(key)
```

Comment:
Seems like we cannot just simply treat `key` for `NoSuchElementException` as same as message in error class.

I think maybe we need to new Exception something like `SparkNoSuchElementException`.

Let me just revert for now, and will open a PR for new Exception, and cover the missing cases here.